File: nextjs/src/type/store.ts
// type/store.ts
export interface StoreBase {
  name: string;
  name_ruby: string;
  postal_code: string;
  prefecture: string;
  street: string;
  address: string;
  building: string;
  phone_number: string;
}

export interface StoreCreate extends StoreBase {}

export interface Store extends StoreBase {
  id: number;
  created_at: Date;
  updated_at: Date;
}

File: fastapi/app/schemas/store.py
from pydantic import BaseModel

class StoreCreateSchema(BaseModel):
    name: str
    name_ruby: str
    postal_code: str
    prefecture: str
    street: str
    address: str
    building: str
    phone_number: str

class StoreUpdateSchema(BaseModel):
    name: str
    name_ruby: str
    postal_code: str
    prefecture: str
    street: str
    address: str
    building: str
    phone_number: str